enterprise/frontend/src/metabase-enterprise/data_editing/tables/EditingBodyCell.tsx

```diff
@@ -0,0 +1,95 @@
+import type { CellContext } from "@tanstack/react-table";
+import {
+  type ChangeEvent,
+  type KeyboardEvent,
+  useCallback,
+  useState,
+} from "react";
+
+import { Input } from "metabase/ui";
+import { isPK } from "metabase-lib/v1/types/utils/isa";
+import type { DatasetColumn, RowValue, RowValues } from "metabase-types/api";
+
+import S from "./TableDataView.module.css";
+import type { RowCellsWithPkValue } from "./types";
+
+interface EditingBodyCellProps<TRow, TValue> {
+  cellContext: CellContext<TRow, TValue>;
+  columns: DatasetColumn[];
+  onCellValueUpdate: (params: RowCellsWithPkValue) => void;
+  onCellEditCancel: () => void;
+}
+
+export const EditingBodyCell = ({
+  cellContext,
+  columns,
+  onCellValueUpdate,
+  onCellEditCancel,
+}: EditingBodyCellProps<RowValues, RowValue>) => {
+  const {
+    getValue,
+    row: { original: rowData },
+    column: { id: columnName },
+  } = cellContext;
+  const initialValue = getValue<RowValue>();
+  const [value, setValue] = useState<RowValue>(initialValue);
+
+  const doCellValueUpdate = useCallback(() => {
+    if (value !== initialValue) {
+      const pkColumnIndex = columns.findIndex(isPK);
```

Comment:
Maybe the api would be a bit cleaner if searching for pk column and row would be in the parent component and here we would pass to onCellValueUpdate only columnName, value, and row index or cell id